src/experimental/Toast/ToastUI.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react'
+import {CheckIcon, AlertIcon, StopIcon, InfoIcon, XIcon} from '@primer/octicons-react'
+
+import type {ForwardRefComponent as PolymorphicForwardRefComponent} from '../../utils/polymorphic'
+import {Spinner, Box, IconButton} from '../..'
+
+type ToastProps = {
+  /**
+   * @default default
+   */
+  variant?: ToastVariants | undefined
+  /**
+   * Setting this to true will allow the user to dismiss this toast.
+   *
+   * @default false
+   */
+  dismissable?: boolean | undefined
+  /**
+   * Sets the roel attribute of the toast, which is used by assistive technologies to communicate status to users.
+   *
+   * - "status" will not be read out of assistive technologies immediately, and won't interupt what the user is currently doing.
+   * - "alert" will be read out immediately, and will interupt what the user is currently doing. Because of this "alert" should be use sparingly.
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alert_role
+   *
+   * @default status
+   */
+  role?: 'status' | 'alert' | undefined
+
+  /**
+   * The `aria-label` to use for the close button.
+   *
+   * @default Dismiss this message
+   */
+  dismissLabel?: string
+}
+
+const variants = ['default', 'success', 'attention', 'danger', 'loading'] as const
+
+export type ToastVariants = (typeof variants)[number]
+
+const icon: Record<ToastVariants, React.ReactNode> = {
+  default: <InfoIcon size="small" />,
+  success: <CheckIcon size="small" />,
+  attention: <AlertIcon size="small" />,
+  danger: <StopIcon size="small" />,
+  loading: <Spinner size="small" />,
+}
+
+export const ToastUI = React.forwardRef((props, ref) => {
+  const {
+    dismissable = false,
+    variant = 'default',
+    role = 'stauts',
+    dismissLabel = 'Dismiss this message',
+    children,
+    ...rest
+  } = props
+
+  if (__DEV__) {
+    // eslint-disable-next-line no-console
+    console.assert(
+      variants.includes(variant),
+      `The variant ${variant} is not supported. Use one of ${variants.join(', ')}`,
+    )
+  }
+
+  return (
+    <Box
+      sx={{display: 'inline-flex', borderRadius: 2, maxWidth: '450px', boxShadow: 'shadow.large', overflow: 'hidden'}}
+      ref={ref}
```

Comment:
will remove this ref for now actually. Initially thought i'd require it for animations, like fading them in and out. But can always add this in later once i figure out how that'll work.